### PR TITLE
Fix Java 8 compatibility issues

### PR DESCRIPTION
**Pull request by edmocosta**
## Release notes

Fixed Java 8 compatibility issues

## What does this PR do?

This PR adds the [--release](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.compile.CompileOptions.html#org.gradle.api.tasks.compile.CompileOptions:release) ([JEP 247](https://openjdk.org/jeps/247)) option to the Gradle's `CompileJava` task, with the same `targetCompatibility` major version value. 

## Why is it important/What is the impact to the user?

The motivation to add this flag is to avoid compatibility issues when the code is compiled using a newer Java version but it's meant to be executed in older JVMs. This issue is explained well in the [blog post](https://www.morling.dev/blog/bytebuffer-and-the-dreaded-nosuchmethoderror/).

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files (and/or docker env variables)
- [ ] I have added tests that prove my fix is effective or that my feature works

## How to test this PR locally

Using this PR source code:
- Download the JDK 8 and set the `LS_JAVA_HOME` to its path
- Enabled the PQ in `config/logstash.yml` by setting the `queue.type: persisted`
- Compile the Java code and run: ```bin/logstash -e "input{stdin{}} output{stdout{}}"```

It should work with no errors.

Repeating all the steps using the `7.17` branch code should produce an error similar to: 
```
{:action=>LogStash::PipelineAction::Create/pipeline_id:main, :exception=>"Java::JavaLang::NoSuchMethodError", :message=>"java.nio.MappedByteBuffer.position(I)Ljava/nio/MappedByteBuffer;", :backtrace=>
```

## Related issues

- Closes https://github.com/elastic/logstash/issues/14883

### CODE REVIEW

**Comment by andsel** [comment]
LGTM,
however the CI is 🔴  because of the error:
```
10:07:29 Execution failed for task ':configureArtifactInfo'.
10:07:29 > could not find the current artifact from the artifact-api https://artifacts-api.elastic.co/v1/versions for 7.17.8
```

If you navigate the URL https://artifacts-api.elastic.co/v1/versions but version `7.17.8` comes out of your branch: https://github.com/edmocosta/logstash/blob/fix-java-8-compatibility-issues/versions.yml#L3

So simply rebase your branch to the latest `7.17`

**Review by andsel — APPROVED** [review]
LGTM
but update your branch to latest `7.17` and wait for 🟢  CI before merge.

**Comment by edmocosta** [comment]
Jenkins test this

**Pull request merged**
